pkg/util/mount/mount_helper_windows.go

```diff
@@ -0,0 +1,63 @@
+// +build windows
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mount
+
+import (
+	"os"
+	"syscall"
+
+	"k8s.io/klog"
+)
+
+const (
+	// failure code is from https://docs.microsoft.com/en-us/windows/desktop/debug/system-error-codes--1300-1699-
+	ERROR_NETNAME_DELETED       = 64
+	ERROR_NETWORK_ACCESS_DENIED = 65
+	ERROR_BAD_NET_NAME          = 67
+	ERROR_LOGON_FAILURE         = 1326
+)
+
+// IsCorruptedMnt return true if err is about corrupted mount point
+func IsCorruptedMnt(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	var underlyingError error
+	switch pe := err.(type) {
+	case nil:
+		return false
+	case *os.PathError:
+		underlyingError = pe.Err
+	case *os.LinkError:
+		underlyingError = pe.Err
+	case *os.SyscallError:
+		underlyingError = pe.Err
+	}
+
+	if ee, ok := underlyingError.(syscall.Errno); ok {
+		errno := int(ee)
+		if errno == ERROR_NETNAME_DELETED || errno == ERROR_NETWORK_ACCESS_DENIED || errno == ERROR_BAD_NET_NAME || errno == ERROR_LOGON_FAILURE {
```

Comment:
also one more question. why is the check that already exists in Linux for underlyingError == syscall.ENOTCONN  not applicable on windows? 
i understand the need to check for additional error codes around ERROR_LOGON_FAILURE for example, but the linux checks should apply as well.

Reply:
hi @michmike this issue is discussed in another PR: https://github.com/kubernetes/kubernetes/pull/74625#issuecomment-469938248, that is: error code is different between Linux and Windows. E.g. on Linux, the SMB password incorrect is `syscall.EACCES` while on Windows, it's `ERROR_LOGON_FAILURE`, they are different error no

Reply:
i agree @andyzhangx , but in Linux we are checking for more error codes than just syscall.EACCES